RecFCCeeCalorimeter/src/components/PhotonIDTool.cpp

```diff
@@ -0,0 +1,352 @@
+#include "PhotonIDTool.h"
+
+// our EDM
+#include "edm4hep/Cluster.h"
+#include "edm4hep/ClusterCollection.h"
+
+#include <fstream>
+
+DECLARE_COMPONENT(PhotonIDTool)
+
+// convert vector data with given shape into ONNX runtime tensor
+template <typename T>
+Ort::Value vec_to_tensor(std::vector<T> &data, const std::vector<std::int64_t> &shape)
+{
+  Ort::MemoryInfo mem_info =
+      Ort::MemoryInfo::CreateCpu(OrtAllocatorType::OrtArenaAllocator, OrtMemType::OrtMemTypeDefault);
+  auto tensor = Ort::Value::CreateTensor<T>(mem_info, data.data(), data.size(), shape.data(), shape.size());
+  return tensor;
+}
+
+PhotonIDTool::PhotonIDTool(const std::string &name,
+			   ISvcLocator *svcLoc)
+    : Gaudi::Algorithm(name, svcLoc)
+{
+  declareProperty("inClusters", m_inClusters, "Input cluster collection");
+  declareProperty("outClusters", m_outClusters, "Output cluster collection");
+}
+
+StatusCode PhotonIDTool::initialize()
+{
+  // Initialize base class
+  {
+    StatusCode sc = Gaudi::Algorithm::initialize();
+    if (sc.isFailure())
+    {
+      return sc;
+    }
+  }
+
+  // read the files defining the model
+  StatusCode sc = readMVAFiles(m_mvaInputsFile, m_mvaModelFile);
+  if (sc.isFailure())
+  {
+    error() << "Initialization of photon ID tool config files not successful!" << endmsg;
+    return sc;
+  }
+
+  // read from the metadata the names of the shape parameters in the input clusters
+  std::vector<std::string> shapeParameters = m_inShapeParameterHandle.get({});
+  debug() << "Variables in shapeParameters of input clusters:" << endmsg;
+  for (const auto &str : shapeParameters) {
+    debug() << str << endmsg;
+  }
+
+  // check if the shape parameters contain the inputs needed for the inference
+  m_inputPositionsInShapeParameters.clear();
+  for (const auto &feature : m_internal_input_names) {
+
+    if (feature == "ecl") {
+      // for the cluster energy, check if we have rawE in decorations
+      // this is for cluster that have been passed through the MVA calibration
+      // otherwise, we will use the energy of the cluster object
+      auto it = std::find(shapeParameters.begin(), shapeParameters.end(), "rawE");
+      if (it != shapeParameters.end())
+	{
+	  int position = std::distance(shapeParameters.begin(), it);
+	  m_inputPositionsInShapeParameters.push_back(position);
+	  info() << "Feature " << feature << " found in position " << position << " of shapeParameters" << endmsg;
+	}
+      else {
+	m_inputPositionsInShapeParameters.push_back(-1);
```

Comment:
`m_inputPositionsInShapeParameters` is `unsigned short int`

Reply:
changed to vector of short int